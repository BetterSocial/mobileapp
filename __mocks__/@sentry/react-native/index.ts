export const addBreadcrumb = jest.fn();
