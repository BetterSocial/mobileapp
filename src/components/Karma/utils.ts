export const renderCircleColor = (karma: number) => {
  switch (true) {
    case karma > 87.5:
      return ['#50BA57', '#C7EECA', '#50BA57'];
    case karma > 75:
      return ['#70BF21', '#B3E57F', '#70BF21'];
    case karma > 62.5:
      return ['#A0C527', '#D1EA82', '#A0C527'];
    case karma > 50:
      return ['#D3D20C', '#F3F370', '#D3D20C'];
    case karma > 37.5:
      return ['#E4B92E', '#FFE591', '#E4B92E'];
    case karma > 25:
      return ['#F4833C', '#F9BE9A', '#F4833C'];
    case karma > 12.5:
    default:
      return ['#DF001F', '#F5B0B9', '#DF001F'];
  }
};
