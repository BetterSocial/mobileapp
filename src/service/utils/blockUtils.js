import Toast from 'react-native-simple-toast';

import {blockAnonymous, blockUser} from '../../service/blocking';

const uiBlockPostAnonymous = async(postId, source, reason, message, callback) => {
    const data = {
        postId,
        source,
        reason,
        message,
        };
    let result = await blockAnonymous(data);
    if (result.code === 201) {
        callback()
        Toast.show(
            'The user was blocked successfully. \nThanks for making BetterSocial better!',
            Toast.LONG,
        );
    } else {
        Toast.show('Your report was filed & will be investigated', Toast.LONG);
    }
}

const uiBlockUser = async(postId, userId, source, reason, message, callback) => {
    const data = {
        userId,
        postId,
        source,
        reason,
        message,
    };
    let result = await blockUser(data);
    if (result.code === 200) {
        callback()
        Toast.show(
            'The user was blocked successfully. \nThanks for making BetterSocial better!',
            Toast.LONG,
        );
    } else {
        Toast.show('Your report was filed & will be investigated', Toast.LONG);
    }
}

const uiBlockDomain = async(domainId, reason, message, source, callback) => {
    const dataBlock = {
        domainId,
        reason,
        message,
        source,
    };
    const result = await blockDomain(dataBlock);
    if (result.code === 200) {
        callback();
        Toast.show(
            'The domain was blocked successfully. \nThanks for making BetterSocial better!',
            Toast.LONG,
        );
    } else {
        Toast.show('Your report was filed & will be investigated', Toast.LONG);
    }
    console.log('result block user ', result);
}


export default {
    uiBlockPostAnonymous,
    uiBlockUser,
    uiBlockDomain
}