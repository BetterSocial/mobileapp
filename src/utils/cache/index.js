import AsyncStorage from '@react-native-async-storage/async-storage';

const CACHE_NAME = 'cache';
export const saveToCache = (cacheKey, jsonData) => {
  if (jsonData && cacheKey && typeof jsonData === 'object' && typeof cacheKey === 'string') {
    AsyncStorage.getItem(CACHE_NAME, (err, result) => {
      if (result && typeof result === 'string') {
        let parseResult = JSON.parse(result);
        parseResult = { ...parseResult, [cacheKey]: jsonData };
        AsyncStorage.setItem(CACHE_NAME, JSON.stringify(parseResult));
      } else {
        const dataSave = {
          [cacheKey]: jsonData,
        };
        AsyncStorage.setItem(CACHE_NAME, JSON.stringify(dataSave));
      }
    });
  }
};

export const getSpecificCache = (cacheKey, callback) => {
  AsyncStorage.getItem(CACHE_NAME, (error, result) => {
    if (result && typeof result === 'string') {
      const newResult = JSON.parse(result);
      if (newResult[cacheKey]) {
        callback(newResult[cacheKey]);
      } else {
        callback(null);
      }
    } else {
      callback(null);
    }
  });
};

export const removeAllCache = () => {
  AsyncStorage.clear();
};
