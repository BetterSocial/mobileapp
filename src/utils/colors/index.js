// https://chir.ag/projects/name-that-color/
export const colors = {
  bondi_blue: '#00ADB5',
  white: '#ffffff',
  black: '#0C0C0C',
  emperor: '#4F4F4F',
  gray: '#69707C',
  gray1: '#9B9FA9',
  lightgrey: '#FFFFFF',
  concrete: '#F2F2F2',
  mine_shaft: '#333333',
  bunting: '#11243D',
  alto: '#E8EBED',
  porcelain: '#ecf0f1',
  silver: '#BDBDBD',
  french_pass: '#b6e4fd',
  pattens_blue: '#ddf2fe',
  blue: '#2F80ED',
  lightblue: '#55C2FF',
  blue1: '#4782D7',
  red: '#FF2E63',
  holytosca: '#003848',
  holytosca30percent: '#0038484D',
  blackgrey: '#69707C',
  redalert: '#FF2E63',
  halfBaked: '#88CDD0',
  tradewind: '#58B1B5',
  elm: '#22878B',
  blockColor: '#FF2E63',
  darkBlue: '#4782D7',
  blueSea10: '#55C2FF',
  greenDark: '#004346',
  babyBlue: '#7B9DCD',

  signed_primary: '#4782D7',
  signed_secondary: '#6295DD',
  anon_primary: '#003848',
  anon_secondary: '#154B5B',
  radical_red: '#FF3466',
  light_silver: '#D2D4DB'
};

export const hexToRgb = (hex, opacity = 1) => {
  const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
  const r = parseInt(result[1], 16);
  const g = parseInt(result[2], 16);
  const b = parseInt(result[3], 16);
  return result ? `rgba(${r},${g},${b},${opacity})` : null;
};

export const checkIsHasColor = (color) => {
  const regex = /^#[0-9a-f]{3,6}$/i;
  return regex.test(color);
};
