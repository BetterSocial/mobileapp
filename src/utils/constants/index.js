const ANALYTICS_SHARE_POST_FEED_ID = 'btn_share';
const ANALYTICS_SHARE_POST_FEED_SCREEN = 'feed_screen_btn_share';
const ANALYTICS_SHARE_POST_PDP_SCREEN = 'pdp_screen_btn_share';
const ANALYTICS_SHARE_POST_PROFILE_ID = 'btn_share';
const ANALYTICS_SHARE_POST_PROFILE_SCREEN = 'profile_screen_btn_share';
const ANALYTICS_SHARE_POST_TOPIC_ID = 'btn_share';
const ANALYTICS_SHARE_POST_TOPIC_SCREEN = 'topic_screen_btn_share';
const CHANNEL_TYPE_PERSONAL = 0;
const CHANNEL_TYPE_GROUP = 1;
const CHANNEL_TYPE_GROUP_LOCATION = 2;
const CHANNEL_TYPE_TOPIC = 3;
const CHANNEL_TYPE_ANONYMOUS = 4;
const DEFAULT_PROFILE_PIC_PATH =
  'https://res.cloudinary.com/hpjivutj2/image/upload/v1680929851/default-profile-picture_vrmmdn.png';
const DEFAULT_TOPIC_PIC_PATH =
  'https://res.cloudinary.com/hpjivutj2/image/upload/v1707840597/qlundmevnhtlwx6vlglo.png';
const DEFAULT_TOPIC_PIC_PATH_ANON =
  'https://res.cloudinary.com/hpjivutj2/image/upload/v1636632905/vdg8solozeepgvzxyfbv.png';
const KEY_ACCESS_TOKEN = 'access_token';
const KEY_ANONYMOUS_TOKEN = 'anonymous_token';
const KEY_REFRESH_TOKEN = 'refresh_token';
const MAX_POLLING_ALLOWED = 4;
const MAX_POLLING_CHARACTER_ALLOWED = 25;
const MIN_POLLING_ALLOWED = 2;
const POST_TYPE_LINK = 2;
const POST_TYPE_POLL = 1;
const POST_TYPE_STANDARD = 0;
const POST_VERB_LINK = 'link';
const POST_VERB_POLL = 'poll';
const POST_VERB_STANDARD = 'tweet';
const FEED_CHAT_KEY = 'feed_chats';

const DISCOVERY_TAB_USERS = 0;
const DISCOVERY_TAB_TOPICS = 1;
const DISCOVERY_TAB_DOMAINS = 2;
const DISCOVERY_TAB_NEWS = 3;

const SOURCE_FEED_TAB = 'feed tab';
const SOURCE_PDP = 'PDP';
const SOURCE_MY_PROFILE = 'my profile';

const PRIVACY_PUBLIC = 'public';
const PRIVACY_PEOPLE_I_FOLLOW = 'people_i_follow';

const ENABLE_DEV_ONLY_FEATURE = true;

const PREFIX_TOPIC = 'topic_';

const POST_CHECK_FEED_NOT_FOUND = 1;
const POST_CHECK_AUTHOR_NOT_FOLLOWING = 2;
const POST_CHECK_FEED_EXPIRED = 3;
const POST_CHECK_AUTHOR_BLOCKED = 4;

const PERMISSION_STATUS_NOT_DEFINED = 0;
const PERMISSION_STATUS_BLOCKED = 1;
const PERMISSION_STATUS_ACCEPTED = 2;
const PERMISSION_STATUS_PENDING = 3;
const CHAT_SIGNED = 'SIGNED';
const CHAT_ANON = 'ANON';

const NavigationConstants = {
  CREATE_POST_SCREEN: 'CreatePost'
};

const POST_VERSION = 2;

const DELETED_MESSAGE_TEXT = 'This message has been deleted';
const CONTEXT_MENU_REPLY = 'Reply';
const CONTEXT_MENU_COPY = 'Copy Message';
const CONTEXT_MENU_DELETE = 'Delete Message';
const MESSAGE_TYPE_REGULAR = 'regular';
const MESSAGE_TYPE_REPLY = 'reply';
const MESSAGE_TYPE_REPLY_PROMPT = 'reply_prompt';
const MESSAGE_TYPE_DELETED = 'deleted';
const FEEDS_DIFF_TO_FETCH = 8;
const IS_ANONYMOUS = 1;
const IS_NOT_ANONYMOUS = 0;

const getPostType = (postTypeInNumber) => {
  switch (postTypeInNumber) {
    case POST_TYPE_LINK:
      return 'post_link';
    case POST_TYPE_POLL:
      return 'post_poll';
    case POST_TYPE_STANDARD:
      return 'post_standard';
    default:
      return '';
  }
};

export {
  getPostType,
  ANALYTICS_SHARE_POST_FEED_ID,
  ANALYTICS_SHARE_POST_FEED_SCREEN,
  ANALYTICS_SHARE_POST_PDP_SCREEN,
  ANALYTICS_SHARE_POST_PROFILE_ID,
  ANALYTICS_SHARE_POST_PROFILE_SCREEN,
  ANALYTICS_SHARE_POST_TOPIC_ID,
  ANALYTICS_SHARE_POST_TOPIC_SCREEN,
  CHANNEL_TYPE_ANONYMOUS,
  CHANNEL_TYPE_GROUP,
  CHANNEL_TYPE_GROUP_LOCATION,
  CHANNEL_TYPE_PERSONAL,
  CHANNEL_TYPE_TOPIC,
  CHAT_ANON,
  CHAT_SIGNED,
  CONTEXT_MENU_COPY,
  CONTEXT_MENU_DELETE,
  CONTEXT_MENU_REPLY,
  DEFAULT_PROFILE_PIC_PATH,
  DEFAULT_TOPIC_PIC_PATH,
  DEFAULT_TOPIC_PIC_PATH_ANON,
  DELETED_MESSAGE_TEXT,
  DISCOVERY_TAB_DOMAINS,
  DISCOVERY_TAB_NEWS,
  DISCOVERY_TAB_TOPICS,
  DISCOVERY_TAB_USERS,
  ENABLE_DEV_ONLY_FEATURE,
  FEEDS_DIFF_TO_FETCH,
  FEED_CHAT_KEY,
  KEY_ACCESS_TOKEN,
  KEY_ANONYMOUS_TOKEN,
  KEY_REFRESH_TOKEN,
  MAX_POLLING_ALLOWED,
  MAX_POLLING_CHARACTER_ALLOWED,
  MESSAGE_TYPE_DELETED,
  MESSAGE_TYPE_REGULAR,
  MESSAGE_TYPE_REPLY,
  MESSAGE_TYPE_REPLY_PROMPT,
  MIN_POLLING_ALLOWED,
  NavigationConstants,
  PERMISSION_STATUS_NOT_DEFINED,
  PERMISSION_STATUS_ACCEPTED,
  PERMISSION_STATUS_BLOCKED,
  PERMISSION_STATUS_PENDING,
  POST_CHECK_AUTHOR_BLOCKED,
  POST_CHECK_AUTHOR_NOT_FOLLOWING,
  POST_CHECK_FEED_EXPIRED,
  POST_CHECK_FEED_NOT_FOUND,
  POST_TYPE_LINK,
  POST_TYPE_POLL,
  POST_TYPE_STANDARD,
  POST_VERB_LINK,
  POST_VERB_POLL,
  POST_VERB_STANDARD,
  POST_VERSION,
  PREFIX_TOPIC,
  PRIVACY_PEOPLE_I_FOLLOW,
  PRIVACY_PUBLIC,
  SOURCE_FEED_TAB,
  SOURCE_MY_PROFILE,
  SOURCE_PDP,
  IS_ANONYMOUS,
  IS_NOT_ANONYMOUS
};
