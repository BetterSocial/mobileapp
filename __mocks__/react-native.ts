import * as RN from 'react-native';

RN.Animated.timing = () => ({
    start: () => jest.fn()
})

module.exports = RN