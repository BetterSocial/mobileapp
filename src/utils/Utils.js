import * as React from 'react';
import SimpleToast from 'react-native-simple-toast';
import { Linking } from 'react-native';
export const sanitizeUrlForLinking = (url) => {
  if (!/^https?:\/\//.test(url)) {
    url = `https://${url}`;
  }

  return url.replace(/(www\.)/, '');
};

export const smartRender = (ElementOrComponentOrLiteral, props, fallback) => {
  if (ElementOrComponentOrLiteral === undefined) {
    ElementOrComponentOrLiteral = fallback;
  }
  if (React.isValidElement(ElementOrComponentOrLiteral)) {
    // Flow cast through any, to make flow believe it's a React.Element
    const element = ElementOrComponentOrLiteral;
    return element;
  }

  // Flow cast through any to remove React.Element after previous check
  const ComponentOrLiteral = ElementOrComponentOrLiteral;
  if (
    typeof ComponentOrLiteral === 'string' ||
    typeof ComponentOrLiteral === 'number' ||
    typeof ComponentOrLiteral === 'boolean' ||
    ComponentOrLiteral == null
  ) {
    return ComponentOrLiteral;
  }
  return <ComponentOrLiteral {...props} />;
};

export const validationURL = (str) => {
  var pattern = new RegExp(
    '^(https?:\\/\\/)?' + // protocol
    '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|' + // domain name
    '((\\d{1,3}\\.){3}\\d{1,3}))' + // OR ip (v4) address
    '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*' + // port and path
    '(\\?[;&a-z\\d%_.~+=-]*)?' + // query string
    '(\\#[-a-z\\d_]*)?$',
    'i',
  ); // fragment locator
  return !!pattern.test(str);
};

export const getUrl = (str) => {
  let url = str.match(/(https?\:\/\/)?([^\.\s]+)?[^\.\s]+\.[^\s]+/gi);
  if (Array.isArray(url)) {
    return url[0];
  } else {
    return str;
  }
};

export const isContainUrl = (str) => {
  let url = getUrl(str);
  return validationURL(url);
};

export const openUrl = (url) => {
  if (url && typeof url === 'string') {
    Linking.canOpenURL(url).then((supported) => {
      if (supported) {
        Linking.openURL(url);
      } else {
        SimpleToast.show('Url is not supported', SimpleToast.SHORT);
      }
    });
  }
};

export const setCapitalFirstLetter = (text) => {
  if (text && typeof text === 'string') {
    const capital = text.charAt(0).toUpperCase();
    const notCapital = text.slice(1);
    return `${capital}${notCapital}`;
  }
};

export const removeWhiteSpace = (txt) => {
  if (txt && typeof txt === 'string') {
    return txt.trim();
  }
};

export const shareUserLink = (username) => {
  return `me.bettersocial.org/u/${username}`
}

export function isEmptyOrSpaces(str) {
  return str === null || str.match(/^ *$/) !== null;
}