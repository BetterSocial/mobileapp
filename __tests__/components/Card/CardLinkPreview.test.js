import * as React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react-native';
import CardLinkPreview from '../../../src/components/Card/CardInLinkPreview';
import TestIdConstant from '../../../src/utils/testId';
import * as utils from '../../../src/utils/Utils';


describe('CardLinkPreview should run correctly', () => {
// date,
//     description,
//     domain,
//     domainImage,
//     image,
//     item,
//     onCardContentPress,
//     onHeaderPress,
//     score,
//     title,
//     url,

    afterEach(cleanup)
    it('should match snapshot', () => {
        const onCardContentPress = jest.fn()
        const onHeaderPress = jest.fn()
        const spy = jest.spyOn(utils, 'sanitizeUrlForLinking')
        const {toJSON, getAllByTestId, getByTestId} = render(<CardLinkPreview domain={'https://detik.com'} onHeaderPress={onHeaderPress} onCardContentPress={onCardContentPress} domainImage='https://detik1.jpg' title='test bro' image='https://detik.jpg' date={'26/08/2022'} />)
        expect(toJSON).toMatchSnapshot()
        fireEvent.press(getByTestId(TestIdConstant.contentLinkHeaderPress))        
        expect(onHeaderPress).toHaveBeenCalled()
        fireEvent.press(getByTestId(TestIdConstant.contentLinkOpenLinkPress))
        expect(spy).toHaveBeenCalled()
        expect(getAllByTestId('contentLinkImageUrlImage')).toHaveLength(1)
        // expect(getAllByTestId(TestIdConstant.iconDomainProfilePicture)).toHaveLength(1)
        const {getAllByTestId: getNoImageId} = render(<CardLinkPreview domain={'https://detik.com'} onHeaderPress={onHeaderPress} onCardContentPress={onCardContentPress} domainImage='https://detik1.jpg' title='test bro' image={null} date={'26/08/2022'}  />)
        expect(getNoImageId('contentLinkImageEmptyStateImage')).toHaveLength(1)
    })
})