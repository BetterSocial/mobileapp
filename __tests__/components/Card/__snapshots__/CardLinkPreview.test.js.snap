// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardLinkPreview should run correctly should match snapshot 1`] = `[Function]`;
